File: java/src/automation-root/automation-framework-rest/src/main/java/com/automation/framework/rest/pages/StudentDetailsRest.java
package com.automation.framework.rest.pages;

import com.automation.api.components.Enrollment;
import com.automation.api.components.Student;
import com.automation.api.pages.StudentDetails;
import com.automation.core.components.FluentRest;
import com.automation.core.logging.Logger;
import com.automation.core.logging.TraceLogger;
import com.automation.framework.rest.components.StudentRest;
import com.google.gson.JsonArray;
import com.google.gson.JsonObject;
import com.google.gson.JsonParser;
import okhttp3.OkHttpClient;
import okhttp3.Request;

import java.io.IOException;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.Objects;
import java.util.stream.Collectors;

public class StudentDetailsRest extends FluentRest implements StudentDetails {
    private int studentId;
    private String fName;

    public StudentDetailsRest(OkHttpClient httpClient) throws IOException {
        this(httpClient, new TraceLogger());
    }

    private StudentDetailsRest(OkHttpClient httpClient, Logger logger) throws IOException {
        this(httpClient, logger, "http://localhost",0);
    }

    public StudentDetailsRest(OkHttpClient httpClient, Logger logger, String baseUrl, int studentId) throws IOException {
        super(httpClient, logger, baseUrl);
        this.studentId = studentId;
        build();
    }

    @Override
    public Enrollment[] enrollments() {
        return new Enrollment[0];
    }

    @Override
    public String firstName() {
        return fName;
    }

    @Override
    public String lastName() {
        return null;
    }

    @Override
    public LocalDateTime enrollmentDate() {
        return null;
    }

    // non interface fields
    public StudentDetailsRest setStudentId(int studentId) throws IOException {
        this.studentId = studentId;
        return new StudentDetailsRest(getHttpClient(), getLogger(), getBaseUrl(), studentId);
    }

    // build pipeline
    private void build() throws IOException {
        // compose route
        String route = String.format("/api/Students/%d", studentId);

        // get all data-rows
        Request request = new Request.Builder().url(getBaseUrl() + "/api/Students").get().build();
        String responseBody = Objects.requireNonNull(getHttpClient().newCall(request).execute().body()).string();
        JsonObject dataRow = new JsonParser().parse(responseBody).getAsJsonObject();

        // set data
        fName = dataRow.get("firstMidName").getAsString();
    }
}

File: csharp/src/AutomationRoot/Automation.Framework.RestApi/Pages/StudentDetailsRest.cs
using Automation.Api.Components;
using Automation.Api.Pages;
using Automation.Core.Components;
using Automation.Core.Logging;
using Automation.Framework.RestApi.Components;
using Newtonsoft.Json.Linq;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net.Http;
using System.Text;
using System.Threading.Tasks;

namespace Automation.Framework.RestApi.Pages
{
    public class StudentDetailsRest : FluentRest, IStudentDetails
    {
        // members
        private string firstMidName;

        public StudentDetailsRest(HttpClient httpClient)
            : this(httpClient, new TraceLogger()) { }

        public StudentDetailsRest(HttpClient httpClient, ILogger logger)
            : this(httpClient, logger, 0) { }

        public StudentDetailsRest(HttpClient httpClient, ILogger logger, int id)
            : base(httpClient, logger)
        {
            Build(id);
        }

        public DateTime EnrollementDate()
        {
            throw new NotImplementedException();
        }

        public IEnrollment[] Enrollments()
        {
            throw new NotImplementedException();
        }

        public string FirstName() => firstMidName;

        public string LastName()
        {
            throw new NotImplementedException();
        }

        // build pipeline
        private void Build(int id)
        {
            // get response
            var response = HttpClient.GetAsync($"/api/Students/{id}").GetAwaiter().GetResult();
            if (!response.IsSuccessStatusCode)
            {
                return;
            }
            var responseBody = response.Content.ReadAsStringAsync().GetAwaiter().GetResult();

            // extract information
            var token = JToken.Parse(responseBody);
            firstMidName = $"{token["firstMidName"]}";
        }
    }
}